README.md

```diff
@@ -89,6 +89,18 @@ functionality and configuration:
 
 We follow the production maturity level defined [here](https://github.com/open-telemetry/community/blob/47813530864b9fe5a5146f466a58bd2bb94edc72/maturity-matrix.yaml#L31)
 
+### Compatibility
+
+When used as a library, the OpenTelemetry Collector attempts to track the currently supported versions of Go, as [defined by the Go team](https://go.dev/doc/devel/release#policy).
+Removing support for an unsupported Go version is not considered a breaking change.
+
+Support for Go versions on the OpenTelemetry Collector will be updated as follows:
+
+1. The first release after the release of a new Go minor version `N` will add build and tests steps for the new Go minor version.
+2. The first release after three months of a new Go minor version `N` will remove support for Go version `N-2`.
```

Comment:
Do we need the 3 months clause? It would be simpler to maintain this without that clause, since you can do both 1 and 2 at once. With the 3 month clause you have to set a reminder to check in 3 months after every Go release. This is a complication that I am not sure is necessary.